### PR TITLE
Adds control for the execution of cell that caused timeout.

### PR DESCRIPTION
**Pull request by melissawm**
This is an (admittedly ugly) temporary fix for the cell timeout problem. If anyone has a better solution, feel free to close this PR :)

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
Good point - I had increased the project-wide cell execution timeout limit to 60 seconds, but that wasn't enough for this cell to run! For this to run on my laptop, I had to bump it to ~300 seconds.

IMO, it would be great if we could keep boilerplate related to handling long execution times to a minimum. In this particular case, one solution might be to reduce the number of epochs from 100 to say 50 or even 40. This would cut the runtime (all of the execution time is due to the training) and would preserve the large-scale features in the final plot which summarizes the general trends in e.g. accuracy vs. training time.

At least this would minimize the need for changes to the code; it would look something like:

````
```{code-cell} ipython3
learning_rate = 0.005
epochs = 40   # Increase the number of epochs to improve accuracy
hidden_size = 100
pixels_per_image = 784
num_labels = 10
```

We've limited the number of epochs here to keep the training time relatively short.
This parameter can be altered to explore the trade-off between training time and
performance of the network.
````

**Comment by melissawm** [comment]
Agreed. What I'm worried about is that we'll have a similar problem in the [Pong tutorial](https://github.com/numpy/numpy-tutorials/pull/35), so I'll maybe do something similar there - just make it a minimal number of epochs/training steps and the user can increase it as needed. Let me fix this one first.

**Comment by melissawm** [comment]
Just for clarification: should we merge this PR first, and @8bitmp3 can fix the authorship later? Another option would be for this change to be incorporated into the authorship fix so it's done in one step. Let me know what you think is best.

**Comment by rossbar** [comment]
It shouldn't matter so much... if this is merged first, then the commit(s) from this PR won't show up in the file-history after #59 is merged. If #59 goes in first, then rebasing this PR on master will give a file-history that has @8bitmp3 as the author of the "root" commit with these added on top.

**Comment by melissawm** [comment]
Merging this one so CI is happy.
